src/scanner/range_iterator.rs

```diff
@@ -0,0 +1,129 @@
+use gcd::Gcd;
```

Comment:
This is some really good code, and the algorithm seems very cool. I'll definitely write about this -- hopefully Hackernews crowd will pick up on it and we get more stars, I can imagine they'd love it :D